hack/e2e.go

```diff
@@ -207,6 +207,11 @@ func run(deploy deployer) error {
 	}
 
 	if *up {
+		// If we tried to bring the Kubemark cluster up, make a courtesy
+		// attempt to bring it down so we're not leaving resources around.
+		if *down {
+			defer xmlWrap("TearDown", deploy.Down)
+		}
 		// Start the cluster using this version.
 		if err := xmlWrap("Up", deploy.Up); err != nil {
 			return fmt.Errorf("starting e2e cluster: %s", err)
```

Comment:
Seems like it makes more sense to call deploy.Down() before return the Up error rather than relying on a defer.

Alternatively we could change the run signature to a tuple such that we can determine whether or not main needs to schedule a teardown (ignoring the error).  For example we could do `return didAttemptUpWithoutSuccessfulDown, err` and if the first value is true call deploy.Down ignoring the error.

However it seems reasonable to expect run() to handle tearing down before returning the error if appropriate.